File: apps/api/app/schemas/prospect.py
from datetime import datetime

from pydantic import BaseModel

from app.schemas.common import ProspectPriority


class ProspectSignal(BaseModel):
    code: str
    label: str
    direction: str
    confidence: float
    evidence: str


class ProspectSignalOutputSchema(BaseModel):
    id: str
    msme_id: str
    prospect_score: int
    priority: ProspectPriority
    likely_credit_need: str
    best_product_fit: str
    next_best_action: str
    outreach_timing: str
    signals: list[ProspectSignal]
    created_at: datetime

File: apps/web/src/lib/schemas/prospect.ts
import { z } from "zod";
import { prospectPrioritySchema } from "./common";

export const prospectSignalSchema = z.object({
  code: z.string(),
  label: z.string(),
  direction: z.string(),
  confidence: z.number(),
  evidence: z.string()
});

export const prospectOutputSchema = z.object({
  id: z.string(),
  msme_id: z.string(),
  prospect_score: z.number(),
  priority: prospectPrioritySchema,
  likely_credit_need: z.string(),
  best_product_fit: z.string(),
  next_best_action: z.string(),
  outreach_timing: z.string(),
  signals: z.array(prospectSignalSchema),
  created_at: z.string()
});

export type ProspectOutput = z.infer<typeof prospectOutputSchema>;
export type ProspectSignal = z.infer<typeof prospectSignalSchema>;
